backend/pkg/db/tables/presets/presets.go

```diff
@@ -0,0 +1,201 @@
+package db_presets
+
+import (
+	"context"
+	"incompetent-hosting-provider/backend/pkg/db"
+	db_instances "incompetent-hosting-provider/backend/pkg/db/tables/instances"
+	"incompetent-hosting-provider/backend/pkg/util"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/rs/zerolog/log"
+)
+
+const TABLE_NAME string = "presets"
+
+type PresetTable struct {
+	Image          db_instances.ImageSpecification `dynamodbav:"image"`
+	Name           string                          `dynamodbav:"name"`
+	Description    string                          `dynamodbav:"description"`
+	PresetId       int                             `dynamodbav:"presetid"`
+	ContainerPorts []int                           `dynamodbav:"containerports"`
+	RequiredEnv    []string                        `dynamodbav:"requiredenv"`
+}
+
+func init() {
+
+	// No setup needed in test run
+	if util.IsTestRun() {
+		return
+	}
+
+	doesTableExist, err := db.DoesTableExist(TABLE_NAME)
+
+	if err != nil {
+		panic("Could not fetch table status")
+	}
+
+	// if table already exists => we are done here
+	if doesTableExist {
+		log.Info().Msgf("Table %v is present", TABLE_NAME)
+		return
+	}
+
+	log.Info().Msgf("Table %v does not exist. Creating...", TABLE_NAME)
+
+	contents := []types.AttributeDefinition{
+		{
+			AttributeName: aws.String("presetid"),
+			AttributeType: types.ScalarAttributeTypeN,
+		},
+	}
+
+	keySchema := []types.KeySchemaElement{
+		{
+			AttributeName: aws.String("presetid"),
+			KeyType:       types.KeyTypeHash,
+		},
+	}
+
+	err = db.CreateTable(TABLE_NAME, contents, keySchema)
+
+	if err != nil {
+		log.Error().Msgf("Could not create table due to an error (%v)", err)
+		return
+	} else {
+		log.Debug().Msgf("Table %v was created.", TABLE_NAME)
+	}
+	insertInitData()
+}
+
+func insertInitData() {
+	if util.IsTestRun() {
+		return
+	}
+
+	conn := db.GetDynamoConn()
+
+	mySQLPreset, err := attributevalue.MarshalMap(PresetTable{
+		PresetId:    1,
+		Name:        "Mysql",
+		Description: "Your very own instance of mysql",
+		Image: db_instances.ImageSpecification{
+			Name: "mysql",
+			Tag:  "latest",
+		},
+		ContainerPorts: []int{3306},
+		RequiredEnv:    []string{"MYSQL_ROOT_PASSWORD"},
```

Comment:
How is the root pasword defined? How do we inform the user about this?

Reply:
Now included in response, see swagger and `docs/endpoint.md`